lib/ansible/modules/network/nxos/nxos_overlay_global.py

```diff
@@ -234,34 +164,26 @@ def main():
 
     warnings = list()
     check_args(module, warnings)
+    result = {'changed': False, 'commands': [], 'warnings': warnings}
 
-    args =  [
-        'anycast_gateway_mac'
-    ]
+    args = PARAM_TO_COMMAND_KEYMAP.keys()
 
-    existing = invoke('get_existing', module, args)
-    end_state = existing
+    existing = get_existing(module, args)
     proposed = dict((k, v) for k, v in module.params.items()
-                    if v is not None and k in args)
+                     if v is not None and k in args)
 
-    result = {}
     candidate = CustomNetworkConfig(indent=3)
-    invoke('get_commands', module, existing, proposed, candidate)
+    get_commands(module, existing, proposed, candidate)
 
     if not module.check_mode:
-        load_config(module, candidate)
-
-    if module._verbosity > 0:
-        end_state = invoke('get_existing', module, args)
-        result['end_state'] = end_state
-        result['existing'] = existing
-        result['proposed'] = proposed
-
-    result['warnings'] = True
+        if candidate:
```

Comment:
You want to swap these conditionals, so that check mode can actually report anything. `load_config()` should be the only thing gated by `if not module.check_mode`, that way the result is correct